src/pages/Sign In/signIn.jsx

```diff
@@ -0,0 +1,52 @@
+import "./signIn.css"
+import {Link} from "react-router-dom"
+
+import { useAuth } from "../../context/Authentication";
+import axios from "axios";
+import { useState } from "react/cjs/react.development";
+
+export function SignIn(){
+    const {login,setLogin}=useAuth();
+    const [loginData,setLoginData]=useState({"email":"","password":""});
+    
+    const onChangeHandler=(e)=>{
+        return setLoginData(loginData=>({...loginData,[e.target.name]:e.target.value}))
+    };
```

Comment:
Instead returning you can put setLoginData(..)

Reply:
will that not result in infinite loop?